var/spack/repos/builtin/packages/bliss/package.py

```diff
@@ -0,0 +1,49 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+class Bliss(Package):
+    """bliss: A Tool for Computing Automorphism Groups and Canonical
+    Labelings of Graphs"""
+
+    homepage = "http://www.tcs.hut.fi/Software/bliss/"
+    url = "http://www.tcs.hut.fi/Software/bliss/bliss-0.73.zip"
+
+    version('0.73', '72f2e310786923b5c398ba0fc40b42ce')
+
+    # Note: Bliss can also be built without gmp, but we don't support this yet
+
+    depends_on("gmp")
+    depends_on("libtool")
+
+    patch("Makefile.spack.patch")
```

Comment:
We could just keep a plain Makefile in the same directory as the package.py. Or we could print a Makefile from the package.py. I'm not saying there's anything wrong with your patch methodology. I've just never seen something like this in Spack before, which always makes me uneasy at first. Does anyone else have any opinions?